client/components/clubs/ClubCard.tsx

```diff
@@ -1,23 +1,137 @@
 import React from 'react'
+import styled from 'styled-components'
 import { useRouter } from 'next/router'
+import { TAGS, TagGroup, TagRow, TagBubble } from './Tag'
 import { Club } from '../../context'
+import { colours } from '../../styles'
+
+// Shared styled components that can probably be factored out later
+const Icon = styled.img<{ size: string }>`
+  height: ${({ size }) => size};
+  object-fit: cover;
+  width: ${({ size }) => size};
+`
+
+const ClubCardContainer = styled.div`
+  border-radius: 8px;
+  overflow: hidden;
+  width: 100%;
+
+  // A subtle shadow around the card
+  box-shadow: 0px 4px 4px 1px ${colours.neutralLight1};
+`
+
+const ClubCardBanner = styled.div<{ bannerImageURL: string }>`
+  background-image: url(${({ bannerImageURL }) => bannerImageURL});
+  background-position: center;
+  background-repeat: no-repeat;
+  background-size: cover;
+  height: 0;
+  padding-top: calc(100% / 3); // 3:1 aspect ratio
+  width: 100%;
+`
+
+const ClubCardContent = styled.div`
+  margin: 24px;
+`
+
+const ClubCardHeader = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  height: 56px;
+  justify-content: space-between;
+`
+
+const ClubCardName = styled.h2`
+  flex-grow: 1;
+  font-size: 20px;
+  margin-bottom: 0;
+  margin-left: 0;
+  margin-right: 16px;
+  margin-top: 0;
+  overflow: hidden;
+  text-overflow: ellipsis;
+
+  // Truncate overflowing text
+  display: -webkit-box;
+  -webkit-line-clamp: 2;
+  -webkit-box-orient: vertical;
+`
+
+const ClubCardDescription = styled.p`
+  height: 64px;
+  overflow: hidden;
+  position: relative;
+  margin-bottom: 24px;
+  margin-top: 16px;
+
+  // Truncate overflowing text with an ellipse
+  display: -webkit-box;
+  text-overflow: ellipsis;
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 3;
+
+  // Fade out overflowing text
+  /*
+  :after {
+    content: '';
+    position: absolute;
+    bottom: 0;
+    right: 0;
+    width: 40%;
+    height: 1.2em;
+    background: linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1) 80%);
+  }
+  */
```

Comment:
If this is commented out should we just delete it?

Reply:
This is here to demonstrate another way we could cut off extra text in the description. This way makes the text fade out, as opposed to the ellipse that the code uses. Forgot to ask design which they prefer.

### Ellipse (current)

![image](https://user-images.githubusercontent.com/30944418/116163709-41a50000-a6c6-11eb-8bfb-4500a5a236f9.png)

### Fade

![image](https://user-images.githubusercontent.com/30944418/116163787-6c8f5400-a6c6-11eb-8a3d-aa1ad31297b7.png)


I didn't like how it looks so I left it out.

cc @CarolinaLi @lamvpham

Reply:
For the record, I got the CSS for fade from here: https://stackoverflow.com/a/22809100

Reply:
Removed the fade code

> "i think ellipses are more common, down to stick with that!" - @CarolinaLi